source/containers/document-pii-detection/parsers/image_analysis/general_ocr/ocr_main.py

```diff
@@ -0,0 +1,386 @@
+import copy
+import math
+import time
+import os
+
+import numpy as np
+import onnxruntime
+from PIL import Image
+import cv2
+
+from .imaug import create_operators, transform
+from .postprocess import build_post_process
+
+cuda_available = False
+
+def sorted_boxes(dt_boxes):
+    """
+    Sort text boxes in order from top to bottom, left to right
+    args:
+        dt_boxes(array):detected text boxes with shape [4, 2]
+    return:
+        sorted boxes(array) with shape [4, 2]
+    """
+    num_boxes = dt_boxes.shape[0]
+    sorted_boxes = sorted(dt_boxes, key=lambda x: (x[0][1], x[0][0]))
+    _boxes = list(sorted_boxes)
+
+    for i in range(num_boxes - 1):
+        if abs(_boxes[i + 1][0][1] - _boxes[i][0][1]) < 10 and (
+            _boxes[i + 1][0][0] < _boxes[i][0][0]
+        ):
+            tmp = _boxes[i]
+            _boxes[i] = _boxes[i + 1]
+            _boxes[i + 1] = tmp
+    return _boxes
+
+class TextClassifier():
+    def __init__(self, model_path):
+        self.weights_path = model_path + 'classifier.onnx'
+
+        self.cls_image_shape = [3, 48, 192]
+        self.cls_batch_num = 30
+        self.cls_thresh = 0.9
+        self.use_zero_copy_run = False
+        postprocess_params = {
+            'name': 'ClsPostProcess',
+            "label_list": ['0', '180'],
+        }
+        self.postprocess_op = build_post_process(postprocess_params)
+
+        self.ort_session = onnxruntime.InferenceSession(self.weights_path, providers=['CUDAExecutionProvider'] if cuda_available else ['CPUExecutionProvider'])
+
+    def resize_norm_img(self, img):
+        imgC, imgH, imgW = self.cls_image_shape
+        h = img.shape[0]
+        w = img.shape[1]
+        ratio = w / float(h)
+        if math.ceil(imgH * ratio) > imgW:
+            resized_w = imgW
+        else:
+            resized_w = int(math.ceil(imgH * ratio))
+        resized_image = np.array(Image.fromarray(img).resize((resized_w, imgH)))
+        #resized_image = cv2.resize(img, (resized_w, imgH))
+        resized_image = resized_image.astype('float32')
+        if self.cls_image_shape[0] == 1:
+            resized_image = resized_image / 255
+            resized_image = resized_image[np.newaxis, :]
+        else:
+            resized_image = resized_image.transpose((2, 0, 1)) / 255
+        resized_image -= 0.5
+        resized_image /= 0.5
+        padding_im = np.zeros((imgC, imgH, imgW), dtype=np.float32)
+        padding_im[:, :, 0:resized_w] = resized_image
+        return padding_im
+
+    def __call__(self, img_list):
+        img_list = copy.deepcopy(img_list)
+        img_num = len(img_list)
+        # Calculate the aspect ratio of all text bars
+        width_list = []
+        for img in img_list:
+            width_list.append(img.shape[1] / float(img.shape[0]))
+        # Sorting can speed up the cls process
+        indices = np.argsort(np.array(width_list))
+
+        cls_res = [['', 0.0]] * img_num
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are initializing a multidimensional array using replication. Replicating a `list` with `*` creates references to the existing objects, not copies, which could introduce bugs. We recommend that you create a `list` of the desired length and then fill in each element with a newly created `list`, or use `list` comprehension. For more information about `list` comprehension, see the following.

[Learn more](https://docs.python.org/3/faq/programming.html#how-do-i-create-a-multidimensional-list)

Reply:
Resolve by using 
```
cls_res = []
for i in range(img_num):
    cls_res.append(['', 0.0])
```